src/widgets/display.coffee

```diff
@@ -41,32 +41,33 @@ class Display extends Widget
   ###
   render
 
-  Replaces the older results in element with
+  Render fist page
   the given
 
   @param {Object} res
   @api public
   ###
-  render: (res) ->
-    context = extend true, res, @extraContext or {}
-    context.is_first = true
-    context.is_last = @controller.status.lastPageReached
-    @addHelpers context
-    @element.html @mustache.render @template, context
-    @trigger("df:rendered", [res])
+  render: (res) -> renderPage(res)
 
+  ###
+  renderNext
 
+  Render next Page
+  @param {Object} res
+  @api public
+  ###
+  renderNext: (res) -> renderPage(res, false)
 
   ###
-  renderNext
+  renderPage
 
   Replaces old results with the new ones in the element
   @param {Object} res
   @api public
   ###
-  renderNext: (res) ->
+  renderPage = (res, is_first = true) ->
```

Comment:
I'm reading the PR diff and I don't know that code well but I'll ask anyway: if `res` holds the response then the page number is there so we don't need `is_first` as an argument:

```
context.is_first = res.page == 1
```

If res is a different thing then forget this...